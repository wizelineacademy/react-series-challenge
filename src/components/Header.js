import React from 'react';
import { NavLink } from 'react-router-dom';

import { ContainerWrapper } from '../styles/ContainerWrapper.style.js';
import {
  HeaderWrapper,
  HeaderContainer,
  HeaderLogo,
  HeaderNavItems
} from '../styles/Header.style.js';

import logo from '../icons/logo.svg';

const Header = () => {
  return (
    <HeaderWrapper>
      <ContainerWrapper>
        <HeaderContainer>
          <NavLink to="/">
            <HeaderLogo src={logo} alt="logo" />
          </NavLink>

          <HeaderNavItems>
            <NavLink to="/favorites">Favorites</NavLink>
          </HeaderNavItems>
        </HeaderContainer>
      </ContainerWrapper>
    </HeaderWrapper>
  )
}

export default Header;
