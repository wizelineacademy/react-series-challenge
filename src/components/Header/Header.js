import React from 'react'
import {
  Header,
  Title,
  Navigation,
  StyledLink
} from './Header.styled'

const HeaderComp = () => {

  return(
    <Header>
      <Title>
        Header
      </Title>
      <Navigation>
        <StyledLink to="/">
          HOME
        </StyledLink>
        <StyledLink to="/favorites">
          FAVORITES
        </StyledLink>
      </Navigation>
    </Header>
  )

}

export default HeaderComp
