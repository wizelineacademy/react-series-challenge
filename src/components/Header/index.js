import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Components
import { MainMenu, SearchBar } from '../';

// Styles
import { HeaderStyled } from './HeaderStyles';

const menuData = [
  {
    section: 'home',
    path: '/'
  },
  {
    section: 'favorites',
    path: '/favorites'
  }
];

class Header extends Component {
  render() {
    const { searchFunction } = this.props;
    return (
      <HeaderStyled>
        <MainMenu menuData={menuData} {...this.props} />
        <SearchBar onChange={searchFunction} />
      </HeaderStyled>
    );
  }
}

Header.propTypes = {
  searchFunction: PropTypes.func.isRequired
};

export default Header;
