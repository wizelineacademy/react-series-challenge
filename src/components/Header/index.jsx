import React from 'react'
import { node } from 'prop-types'

import {
  ChildrenWrapper,
  NavWrapper,
  Option,
  OptionsWrapper,
  StyledH1,
  StyledLink,
  Title,
} from './Header.style'
import { paths } from '../../utils'

const Header = ({ children }) => (
  <div>
    <NavWrapper>
      <OptionsWrapper>
        <Option>
          <StyledLink to={paths.HOME}>
            Search
          </StyledLink>
        </Option>
        <Option>
          <StyledLink to={paths.FAVORITES}>
            Favorites
          </StyledLink>
        </Option>
      </OptionsWrapper>
      <Title>
        <StyledH1>React - Giphy</StyledH1>
      </Title>
      <ChildrenWrapper>
        {children}
      </ChildrenWrapper>
    </NavWrapper>
  </div>
)

Header.propTypes = {
  children: node,
}
Header.defaultProps = {
  children: null
}
export default Header
