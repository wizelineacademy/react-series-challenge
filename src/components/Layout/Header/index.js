import React from 'react';
import { Link } from 'react-router-dom';
import logo from '../../../assets/icons/logo.jpg';
import { Row, Col } from '../../Grid';
import StyledHeader from './styles';

const Header = ({ children }) => {
  return (
    <StyledHeader>
      <Row>
        <Col span={12}>
          <Link
            to="/"
          >
            <img className="logo" src={logo} alt="logo" />
            <h1>Welcome to GiPHY</h1>
          </Link>
        </Col>
      </Row>
    </StyledHeader>
  );
};

export default Header;
