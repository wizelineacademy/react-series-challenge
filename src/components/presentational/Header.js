import React from 'react';
import styled from 'styled-components';

const HeaderStyled = styled.div`
  background-color: #20232a;
  h1 {
    color: #fff;
    font-size: 30px;
    text-align: center;
    padding: 30px;
  }
`;

const Header = () => (
  <HeaderStyled>
    <h1>Giphy Collection</h1>
  </HeaderStyled>
);

export default Header;
