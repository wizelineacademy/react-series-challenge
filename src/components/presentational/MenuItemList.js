import React from 'react';
import MenuItem from './MenuItem';

const MenuItemList = ({ items }) => (
  items.map((item, index) => (
    <MenuItem key={item.route} section={item}/>
  ))
);

export default MenuItemList;
