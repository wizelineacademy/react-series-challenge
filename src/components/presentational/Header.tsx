import * as React from 'react';
import AppNav from '../AppNav';
import { Container } from './Layout';

const Header = (props: any) => {
    return <AppNav />;
}

export default Header;