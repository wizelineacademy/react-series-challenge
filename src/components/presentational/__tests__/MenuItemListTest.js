import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'

import MenuItemList from '../MenuItemList';
import Header from '../Header';

describe('Menu Item List', () => {
  it('should render without crashing', () => {
    const sections = [
      { route: 'test', title: 'Test' , component: Header },
    ];
    const testMenuItemList = shallow(<MenuItemList items={sections} />);    

    expect(shallowToJson(testMenuItemList)).toMatchSnapshot()
  })
})


