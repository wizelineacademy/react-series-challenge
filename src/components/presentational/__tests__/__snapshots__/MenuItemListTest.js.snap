// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Item List should render without crashing 1`] = `
Array [
  <MenuItem
    key="test"
    section={
        Object {
            "component": [Function],
            "route": "test",
            "title": "Test",
          }
    }
/>,
]
`;
